### PR TITLE
[NVPTX] disable combine-mad test for newer ptxas

### PR DESCRIPTION
**Pull request by AlexMaclean**

### CODE REVIEW

**Comment by AlexMaclean** [comment]
@Artem-B, this addresses your request from https://github.com/llvm/llvm-project/pull/93628#discussion_r1621453827. Can you clarify what version of ptxas you're using and what error you're seeing?

**Comment by Artem-B** [comment]
> @Artem-B, this addresses your request from [#93628 (comment)](https://github.com/llvm/llvm-project/pull/93628#discussion_r1621453827). Can you clarify what version of ptxas you're using and what error you're seeing?

The error:
```
[  4] ; RUN: %if ptxas %{ llc < %s -mtriple=nvptx -mcpu=sm_20 -O1 | %ptxas-verify %} [FAIL]
 llc < /build/work/ea8b3a3717bb294ac025e537a8aabcc8890c/google3/runfiles/google3/third_party/llvm/llvm-project/llvm/test/CodeGen/NVPTX/combine-mad.ll -mtriple=nvptx -mcpu=sm_20 -O1 | /build/work/ea8b3a3717bb294ac025e537a8aabcc8890c/google3/runfiles/google3/third_party/gpus/cuda_12_0/bin/ptxas -arch=sm_60 -c -o /dev/null - 
ptxas warning :  64 Bit host architecture (--machine) being used mismatches with .address_size of 32 bits
ptxas fatal   :  32-Bit compilation is no longer supported
Command failed: exit status 255
```

File: llvm/test/CodeGen/NVPTX/combine-mad.ll [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
 ; RUN: llc < %s -mtriple=nvptx -mcpu=sm_20 -O1 | FileCheck %s
 ; RUN: llc < %s -mtriple=nvptx64 -mcpu=sm_20 -O1 | FileCheck %s
-; RUN: %if ptxas %{ llc < %s -mtriple=nvptx -mcpu=sm_20 -O1 | %ptxas-verify %}
-; RUN: %if ptxas %{ llc < %s -mtriple=nvptx64 -mcpu=sm_20 -O1 | %ptxas-verify %}
+; RUN: %if ptxas && !ptxas-12.0 %{ llc < %s -mtriple=nvptx -mcpu=sm_20 -O1 | %ptxas-verify %}
+; RUN: %if ptxas && !ptxas-12.0 %{ llc < %s -mtriple=nvptx64 -mcpu=sm_20 -O1 | %ptxas-verify %}
```
Comment (Artem-B) [inline llvm/test/CodeGen/NVPTX/combine-mad.ll:5]: This line does not need the change. 64-bit variant works everywhere.

**Review by Artem-B — APPROVED** [review]
The change should apply to line 4 only. The change itself is fine.

You can (and should run LLVM tests with `LLVM_PTXAS_EXECUTABLE=` set to point to `ptxas` from a recent CUDA version.

File: llvm/test/CodeGen/NVPTX/combine-mad.ll (hunk 1 above)
Comment (AlexMaclean) [reply llvm/test/CodeGen/NVPTX/combine-mad.ll]: Fixed

**Pull request merged**